torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -244,13 +247,84 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                         )
 
                     node.replace_all_uses_with(replacement_proxy.node)
-
                     graph.erase_node(node)
 
-    # DCE symbols (which are guaranteed to be pure) only
-    for proxy in reversed(expr_to_sym_proxy.values()):
-        if len(proxy.node.users) == 0 and proxy.node.op != "placeholder":
-            graph.erase_node(proxy.node)
+    # Now do one more pass that specializes all symfloats we didn't manage
+    # to tensorify away.
+    nodes = list(graph.nodes)
+    for i, node in reversed(list(enumerate(nodes))):
+        if node.op == "output" or node.op == "placeholder":
+            continue
+
+        with graph.inserting_before(nodes[i]):
+            if len(node.users) == 0 and not node.is_impure():
+                graph.erase_node(node)
+                continue
+
+            args = []
+            kwargs: Dict[str, Any] = {}
+            transform = False
+            for a in node.args:
+                if (
+                    isinstance(a, fx.Node)
+                    and (zf := a.meta.get("val")) is not None
```

Comment:
nit: zf naming here is a bit misleading, since typically the zf here won't be a symbol, it will be a generic sympy expression